userspace/falco/falco_outputs.cpp

```diff
@@ -274,8 +278,25 @@ inline void falco_outputs::push(const ctrl_msg& cmsg)
 #ifndef __EMSCRIPTEN__
 	if (!m_queue.try_push(cmsg))
 	{
-		fprintf(stderr, "Fatal error: Output queue reached maximum capacity. Exiting.\n");
-		exit(EXIT_FAILURE);
+		switch (m_outputs_queue_recovery)
+		{
+		case falco_common::RECOVERY_EXIT:
+			falco_logger::log(LOG_ERR, "Fatal error: Output queue out of memory. Exiting ...");
+			exit(EXIT_FAILURE);
+		case falco_common::RECOVERY_EMPTY:
+			m_outputs_queue_num_drops += m_queue.size();
```

Comment:
Stating the docs of `falco_outputs`:
```
All methods in this class are thread-safe. The output framework supports a multi-producer model where messages are stored in a queue and consumed by each configured output asynchrounously.
```

Since `m_queue` is a thread-safe object as guaranteed by TBB, I'd say `m_outputs_queue_num_drops` should be atomic here.

Reply:
good catch!

Reply:
Also realized I think when counting the number of drops for the empty queue case I need to add  +1 will do this later.